packages/runner/lib/sdk/persist.ts

```diff
@@ -0,0 +1,312 @@
+import https from 'node:https';
+import type { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import axios, { AxiosError, isAxiosError } from 'axios';
+import { getPersistAPIUrl } from '@nangohq/shared';
+import { getUserAgent } from '@nangohq/node';
+import { httpRetryStrategy, retryWithBackoff, Ok, Err } from '@nangohq/utils';
+import { logger } from '../logger.js';
+import type { Result } from '@nangohq/utils';
+import type { CursorOffset, DeleteRecordsSuccess, GetCursorSuccess, MergingStrategy, PostRecordsSuccess, PutRecordsSuccess } from '@nangohq/types';
+
+export interface PersistApi {
```

Comment:
persist related code in runner-sdk `NangoSync` was starting to get out of control. Moving all the persist related code to its own file